rust/stackablectl/src/cmds/debug.rs

```diff
@@ -0,0 +1,380 @@
+    },
+};
+use termion::{raw::IntoRawMode, terminal_size};
+use tokio::{
+    io::{unix::AsyncFd, AsyncRead},
+    signal::unix::SignalKind,
+};
+use tracing::{error, info, info_span, warn, Instrument};
+
+use crate::cli::Cli;
+
+#[derive(Debug, Snafu)]
+pub enum CmdError {
+    #[snafu(display("failed to create Kubernetes client"))]
+    KubeClientCreate { source: kube::Error },
+    #[snafu(display("failed to get {pod}"))]
+    GetPod {
+        source: kube::Error,
+        pod: ObjectRef<Pod>,
+    },
+    #[snafu(display("{pod} has no container {container:?}"))]
+    FindTemplateContainer {
+        pod: ObjectRef<Pod>,
+        container: String,
+    },
+    #[snafu(display("failed to create ephemeral debug container {container:?} on {pod}"))]
+    CreateDebugContainer {
+        source: kube::Error,
+        pod: ObjectRef<Pod>,
+        container: String,
+    },
+    #[snafu(display("debug container {container:?} on {pod} never became ready"))]
+    AwaitDebugContainerReadiness {
+        source: kube::runtime::wait::Error,
+        pod: ObjectRef<Pod>,
+        container: String,
+    },
+    #[snafu(display("failed to get status of debug container {container:?} on {pod}"))]
+    FindDebugContainerStatus {
+        pod: ObjectRef<Pod>,
+        container: String,
+    },
+    #[snafu(display("failed to attach to container {container:?} on {pod}"))]
+    AttachContainer {
+        source: kube::Error,
+        pod: ObjectRef<Pod>,
+        container: String,
+    },
+    #[snafu(display("failed to enable raw local TTY input"))]
+    SetRawTtyMode { source: std::io::Error },
+    #[snafu(display("failed to turn stdin async"))]
+    AsyncifyStdin { source: std::io::Error },
+    #[snafu(display("failed to initialize AsyncFd for stdin"))]
+    AsyncFdStdin { source: std::io::Error },
+    #[snafu(display("container has no terminal size channel"))]
+    NoTerminalSizeChannel,
+    #[snafu(display("failed to read terminal size"))]
+    GetTerminalSize { source: std::io::Error },
+    #[snafu(display("failed to update terminal size"))]
+    UpdateTerminalSize { source: mpsc::SendError },
+    #[snafu(display("container has no stdin channel"))]
+    NoStdinChannel,
+    #[snafu(display("container has no stdout channel"))]
+    NoStdoutChannel,
+    #[snafu(display("failed to forward stdin to container"))]
+    ForwardStdin { source: std::io::Error },
+    #[snafu(display("failed to forward stdout from container"))]
+    ForwardStdout { source: std::io::Error },
+}
+type Result<T, E = CmdError> = std::result::Result<T, E>;
+
+#[derive(Debug, Args)]
+pub struct DebugArgs {
+    /// The namespace of the Pod being debugged
+    #[clap(long, short)]
+    namespace: Option<String>,
+    /// The Pod to debug
+    pod: String,
+    /// The target container to debug
+    ///
+    /// Volumes and environment variables will be copied from this container.
+    #[clap(long, short)]
+    container: String,
+    /// The debug container image
```

Comment:
suggestion: Does it maybe make sense to provide a default image suited for debugging? Or at least we could provide a suggestion for which image to use.

Because I'm already wondering which image you sued while developing this feature.

Reply:
Mostly alpine or archlinux. Honestly, this is a bit of an open question. I'm almost leaning towards the best default "for now" being the target container's image since it at least has the product tools installed.

Reply:
Should also look into running the debug container as root instead...

Reply:
> Mostly alpine or archlinux. Honestly, this is a bit of an open question. I'm almost leaning towards the best default "for now" being the target container's image since it at least has the product tools installed.

Alright, then we should maybe add this as a hint to the arguments help text. It would definitely help to remove some of the guesswork involved when choosing a container image.

Regarding using a default value: I would leave it up to you.

Reply:
Default/docs: 9d26417dd1a7c1223b579d7099efefff8ebf0cbb
Root: bc4b32a6194c6082d249daf5439fd83628bbc2cd